src/parser.c

```diff
@@ -67,30 +85,39 @@ union transaction_type_id_t convert_type_id_to_type(uint32_t raw_type_id, enum c
     switch (chain) {
     case CHAIN_X:
         switch (raw_type_id) {
-        case TRANSACTION_X_CHAIN_TYPE_ID_BASE:
-        case TRANSACTION_X_CHAIN_TYPE_ID_IMPORT:
-        case TRANSACTION_X_CHAIN_TYPE_ID_EXPORT:
+        case TRANSACTION_X_CHAIN_TYPE_ID_BASE: // overlaps with C
+        case X_TXTS:
           return (union transaction_type_id_t) { .x = raw_type_id };
+        case P_TXTS:
+        case C_TXTS:
+          REJECT("Blockchain ID did not match expected value for network ID");
         default:
           ; // error at end
         }
         break;
     case CHAIN_P:
         switch (raw_type_id) {
-        case TRANSACTION_P_CHAIN_TYPE_ID_ADD_VALIDATOR:
-        case TRANSACTION_P_CHAIN_TYPE_ID_ADD_DELEGATOR:
-        case TRANSACTION_P_CHAIN_TYPE_ID_IMPORT:
-        case TRANSACTION_P_CHAIN_TYPE_ID_EXPORT:
+        case P_TXTS:
           return (union transaction_type_id_t) { .p = raw_type_id };
+        case TRANSACTION_X_CHAIN_TYPE_ID_BASE: // overlaps with C
+        case X_TXTS:
+        case C_TXTS:
+          REJECT("Blockchain ID did not match expected value for network ID");
         default:
           ; // error at end
         }
         break;
     case CHAIN_C:
         switch (raw_type_id) {
-        case TRANSACTION_C_CHAIN_TYPE_ID_IMPORT:
-        case TRANSACTION_C_CHAIN_TYPE_ID_EXPORT:
+        case TRANSACTION_C_CHAIN_TYPE_ID_IMPORT: // overlaps with X
+        case C_TXTS:
           return (union transaction_type_id_t) { .c = raw_type_id };
+        // case P_TXTS: // nicer error below
```

Comment:
is this commented-out line because of the overlap?

Reply:
Yes, exactly